modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -390,12 +390,28 @@ else if (value instanceof Date[]) {
 				}
 				else if (value instanceof Object[]) {
 					Object[] values = (Object[])value;
+					String type = field.getType();
 
-					for (int i = 0; i < values.length; i++) {
-						sb.append(values[i]);
+					if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
+						for (int i = 0; i < values.length; i++) {
+							String valueString = _getSortableValue(
+								ddmStructure.getDDMFormField(field.getName()),
+								locale, values[i].toString());
 
-						if (i < (values.length - 1)) {
-							sb.append(StringPool.SPACE);
+							sb.append(HtmlUtil.extractText(valueString));
+
+							if (i < (values.length - 1)) {
+								sb.append(StringPool.SPACE);
+							}
+						}
+					}
+					else {
+						for (int i = 0; i < values.length; i++) {
+							sb.append(values[i]);
+
+							if (i < (values.length - 1)) {
+								sb.append(StringPool.SPACE);
+							}
```

Comment:
Hi @kristofloneck,

Do you think we can reuse the existing `for` instead?

Something like:
```
for (int i = 0; i < values.length; i++) {
    if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
        String valueString = _getSortableValue(
            ddmStructure.getDDMFormField(field.getName()),
            locale, values[i].toString());

        sb.append(HtmlUtil.extractText(valueString));
    }
    else {
        sb.append(values[i]);
    }

    if (i < (values.length - 1)) {
	sb.append(StringPool.SPACE);
    }
}
```

Reply:
Hi,
wouldn't that have a worse performance as the if statement is evaluated in every cycle?

Reply:
For sure, @kristofloneck, it'll be one more check to be processed in each `for` loop iteration although it should have a cheap cost. :slightly_smiling_face:

Instead of my suggestion, do you see another way for we prevent code duplication (`for` loop)? This was my concern with my suggestion to you.

Let's find out the best code together! :smile:

Reply:
The best solution I can image is introduction of a new boolean variable before your (for loop) and in the If statement check the value of the boolean, this way we minimalize the performance cost in the code.

````
boolean isItRichText = type.equals(DDMFormFieldTypeConstants.RICH_TEXT);
for (int i = 0; i < values.length; i++) {
    if (isItRichText) {
        String valueString = _getSortableValue(
            ddmStructure.getDDMFormField(field.getName()),
            locale, values[i].toString());

        sb.append(HtmlUtil.extractText(valueString));
    }
    else {
        sb.append(values[i]);
    }

    if (i < (values.length - 1)) {
	sb.append(StringPool.SPACE);
    }
}
````

Reply:
Hi @kristofloneck,

Your point here is that check for a primitive type inside a loop has a better performance than the previous check, right?

If so it makes sense to me, my only concern is that BChan usually doesn't like/merge variables created to be used just once. We can try to move on with your last proposal, but we should add a comment in the pull request to make clear to BChan why we're creating a variable to be used just once. Ideally, we should also add a reference explaining why it has a better performance. :slightly_smiling_face: 

Lastly, I'd recommend we name the boolean variable as `isRichText` instead, since this is a more common pattern in Liferay code.

Thanks!

Reply:
Hi @natocesarrego 
I did the changes as you requested.
Best of all
Kristof.